Rep lookup

turino opened the pull request:
fetch reps individually. (Bulk fetch coming soon)

Rio517 commented on app/models/district.rb:
```diff
@@ -13,6 +13,7 @@ class District < ActiveRecord::Base
   belongs_to :state
   has_and_belongs_to_many :zip_codes
   has_and_belongs_to_many :campaigns
+  has_one :legislator#, dependent: :delete
```
There may not be so much value in this, but you might consider naming these to `has_one :representative, class_name: "Legislator`"  Feel free to ignore.

Rio517 commented on app/models/legislator.rb:
```diff
@@ -0,0 +1,54 @@
+
+  private
+
+  def self.sunlight_params(bioguide_id, district, state, senate_class)
```
I'd consider making `sunlight_params` a helper method in the `Integration::Sunlight` class.  Then you might have a public `Integration::Sunlight.query_legislator_by_params({})` and have it call a `Integration::Sunlight.parse_params` before passing to `Integration::Sunlight.get_legislator`.

Other notes:
* I'm not sure these all need to be class methods, but no biggy.
* we should set up a method to fetch_all legislators. It would be good if it could parse a few at a time, but I can see how you need to replace them with the target IDs.
* `replace_state_and_district`, `update_stats` and `stats` are a bit vague.  Maybe something like `find_location_associations` `location_associations` - I'm not sure what's most appropriate.

Rio517 commented:
Other than the two comments above this is fine.  Do you mind addressing those then I can merge?

Also, now that these are in place we should popualate these in seeds.rb. Later, we should setup some local version so seeding the DB doesn't require so many https requests.

turino commented:
@Rio517 I think I addressed all the issues you raised. I also added bulk fetching of legislators and switched campaigns to target legislators instead of districts.

Still need to update districts_controller to identify target senators.

Also, might want to add logic to ensure only one rep per district and 2 senators per state.
